core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -447,6 +447,32 @@ impl RocksDBStore {
         Ok(rocks_db_store)
     }
 
+    /// Creates a readonly [`RocksDBStore`] that allows reading from the store while some other
+    /// process is writing to it. Any write operation that happens against a read-only store leads
+    /// to a panic.
```

Comment:
Could we explain _why_ we have this? / why we need to keep this?

As I don't believe any node developers think of the node repo as being a library, so we typically just use `pub` for `pub (crate)` and might just delete this bit of code thinking it's unused :D

Reply:
Good point! Will add to this doc comment explaining why its needed and why it has no usages in the node code.